scheduler/reconcile.go

```diff
@@ -321,16 +320,20 @@ func (a *allocReconciler) cancelUnneededDeployments() {
 func (a *allocReconciler) handleStop(m allocMatrix) {
 	for group, as := range m {
 		as = filterByTerminal(as)
-		untainted, migrate, lost := as.filterByTainted(a.taintedNodes)
-		a.markStop(untainted, "", allocNotNeeded)
-		a.markStop(migrate, "", allocNotNeeded)
-		a.markStop(lost, structs.AllocClientStatusLost, allocLost)
 		desiredChanges := new(structs.DesiredUpdates)
-		desiredChanges.Stop = uint64(len(as))
+		desiredChanges.Stop = a.stopAllocSet(as)
 		a.result.desiredTGUpdates[group] = desiredChanges
 	}
 }
 
+func (a *allocReconciler) stopAllocSet(set allocSet) uint64 {
+	untainted, migrate, lost := set.filterByTainted(a.taintedNodes)
+	a.markStop(untainted, "", allocNotNeeded)
+	a.markStop(migrate, "", allocNotNeeded)
+	a.markStop(lost, structs.AllocClientStatusLost, allocLost)
+	return uint64(len(untainted) + len(migrate) + len(lost))
```

Comment:
This sum of lengths doesn't look to be equivalent to the original `len(as)` (I'm pretty sure `filterByTainted` removes items from the possible set)